api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -27,11 +27,24 @@ message Route {
   RouteAction route = 2 [(validate.rules).message.required = true, (gogoproto.nullable) = false];
 }
 
-// [#comment:next free field: 2]
+// [#comment:next free field: 4]
 message RouteMatch {
-  // If specified, the route must exactly match the request method name. As a special case, an
-  // empty string matches any request method name.
-  string method = 1;
+  oneof match_specifier {
```

Comment:
This will allow Thrift requests to be routed based on either method name or service name. @fishcakez and @rgs1, do you think we'll need to be able to route based on the combination of both, e.g. `service_name=X and method_name=Y`?

Reply:
FWIW, you can do this already with `method_name=service:method`

Reply:
Cool, thanks. That solves my one concern.